src/Libraries/Inscriptions/Envelope.ts

```diff
@@ -60,16 +61,26 @@ export class Envelope {
    * @param tx - Raw bitcoin transaction to search for an inscription envelope
    */
   static fromTransaction(tx: RawTransaction) {
-    const [data, oip] = getEnvelopeDataFromTx(tx) ?? [];
-    if (data) {
-      return new Envelope(tx.txid, data, oip);
+    const envelopes = getEnvelopesDataFromTx(tx);
+    if (envelopes) {
+      return envelopes.map(([data, oip], index) => {
+        if (data) {
+          return new Envelope(tx.txid, data, oip, index);
+        }
+        return undefined;
```

Comment:
maybe we need to `.filter` `undefined` out? because `[1,2,undefined,4]` is still a valid array